.github/workflows/compile-examples.yml

```diff
@@ -1,20 +1,87 @@
 name: Compile Examples
-on: [push, pull_request]
+
+# See: https://docs.github.com/en/free-pro-team@latest/actions/reference/events-that-trigger-workflows
+on:
+  push:
+    paths:
+      - ".github/workflows/compile-examples.yml"
+      - "examples/**"
+      - "src/**"
+  pull_request:
+    paths:
+      - ".github/workflows/compile-examples.yml"
+      - "examples/**"
+      - "src/**"
+  schedule:
+    # Run every Tuesday at 8 AM UTC to catch breakage caused by changes to external resources (libraries, platforms).
+    - cron: "0 8 * * TUE"
+  workflow_dispatch:
+  repository_dispatch:
+
 jobs:
- build:
-   runs-on: ubuntu-latest
-
-   strategy:
-     matrix:
-       fqbn: [
-         "arduino:avr:uno",
-         "arduino:samd:mkrzero"
-       ]
-
-   steps:
-     - uses: actions/checkout@v1
-       with:
-         fetch-depth: 1
-     - uses: arduino/actions/libraries/compile-examples@master
-       with:
-         fqbn: ${{ matrix.fqbn }}
+  build:
+    name: ${{ matrix.board.fqbn }}
+    runs-on: ubuntu-latest
+
+    env:
+      SKETCHES_REPORTS_PATH: sketches-reports
+
+    strategy:
+      fail-fast: false
+
+      matrix:
+        board:
+          - fqbn: arduino:avr:nano
```

Comment:
Since this library was written to support the [Arduino MKR 485 Shield](https://store.arduino.cc/arduino-mkr-485-shield) I'd argue that CI only need to run for MKR form factor boards (or boards which expose MKR headers).

Reply:
The reason I did more is because the description implies that it was intended to be used for more than only the MKR 485 Shield:
https://github.com/arduino-libraries/ArduinoRS485/blob/master/README.adoc
>Enables sending and receiving data using the RS485 standard with RS485 shields, like the MKR 485 Shield.
This library supports the Maxim Integrated MAX3157 and equivalent chipsets.

And even the name doesn't follow the "Arduino_MKR485" format I expect from the MKR hardware-specific boards

And the library.properties claims universal compatibility:
https://github.com/arduino-libraries/ArduinoRS485/blob/3a91ebc2b0fc5974e6830ffb64f29667b3ee9db5/library.properties#L9

I'm happy to restrict the CI, but I would suggest that the documentation and metadata be made more clear regarding its compatibility so users don't end up wasting time by thinking that this is the Arduino library for RS-485.

Reply:
I agree that the name is misleading (maybe even intentionally so). For now I'd restrict the library to `samd`.

Reply:
I have updated the examples compilation CI workflow to only compile for boards with MKR headers: https://github.com/arduino-libraries/ArduinoRS485/compare/794dbcc139ae9983c504445e2fb780578491d102..75a78f1611cb53e1de60a5fe2d2b860f1d6b24f2